datafusion/optimizer/src/optimizer.rs

```diff
@@ -225,6 +225,14 @@ impl Optimizer {
                 let result = rule.try_optimize(&new_plan, optimizer_config);
                 match result {
                     Ok(Some(plan)) => {
+                        if plan.schema() != new_plan.schema() {
+                            return Err(DataFusionError::Internal(format!(
```

Comment:
@jackwener 
Is it possible that the check can not always pass in case that some of the optimization rules might change the nullability of the Schema ? 
For example and Column Expression was replaced by a Literal Constant due to Constant propagation ?

Reply:
Indeed, we may need to relax restrictions on some conditions, such as `nullable`.